### PR TITLE
Rename model_mode, break up KV cache impl and test Model

### PR DESCRIPTION
**Pull request by rwitten**

### CODE REVIEW

File: MaxText/tests/model_test.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+import pyconfig
+
+from layers import models
+
+Mesh = jax.sharding.Mesh
+
+
+class ModelTest(unittest.TestCase):
+  """Test for the Attention """
```
Comment (khatwanimohit) [inline MaxText/tests/model_test.py:37]: nit: Test for the Model

File: MaxText/tests/model_test.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      self.assertTrue(
+        prefill_logits_idx.shape == output_logits.shape
+      )
+      self.assertTrue(
+        jax.numpy.allclose(
+            prefill_logits_idx, output_logits, rtol=1e-01, atol=1e-01, equal_nan=False
+        )
+      )
+
+    pass
```
Comment (khatwanimohit) [inline MaxText/tests/model_test.py:133]: nit: remove pass

**Review by khatwanimohit — APPROVED** [review]
LGTM! Just some nits

File: MaxText/tests/model_test.py (hunk 1 above)
Comment (rwitten) [reply MaxText/tests/model_test.py:37]: done

File: MaxText/tests/model_test.py (hunk 2 above)
Comment (rwitten) [reply MaxText/tests/model_test.py]: done

**Comment by rwitten** [comment]
Closing in favor of https://github.com/google/maxtext/pull/308

**Pull request closed without merging**